File: AdminPanel/DataAccess/Entities/Task.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace StudyBuddySAP.DataAccess.Entities

{
    [BsonIgnoreExtraElements]
    public class Task

    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public ObjectId _id { get; set; }
        public string? title { get; set; }
        public string? description { get; set; }
        public bool completed { get; set; }
        public DateTime created_at { get; set; }
        public ObjectId created_by { get; set; }
        public DateTime? due_at { get; set; }
        public ObjectId? group_id { get; set; }
        public ObjectId[] users_assigned { get; set; }
        public int __v { get; set; }
    }
}

File: backend/model/Task.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const tasksSchema = new Schema({
    title: {
        type: String,
        required: true
    },

    description: {
        type: String
    },

    completed: {
        type: Boolean,
        default: false
    },

    created_at: {
        type: Date,
        default: Date.now()
    },

    created_by: {
        type: Schema.Types.ObjectId,
        ref: 'User'
    },

    group_id: {
        type: Schema.Types.ObjectId,
        ref: 'Group',
    },

    due_at: {
        type: Date,
        default: null
    },
    
    users_assigned: [{
        type: Schema.Types.ObjectId,
        ref: 'User'
    }]
});

module.exports = mongoose.model('Task', tasksSchema);